forest/shared/src/logger/mod.rs

```diff
@@ -2,15 +2,37 @@
 // SPDX-License-Identifier: Apache-2.0, MIT
 
 use crate::cli::{cli_error_and_die, LogValue};
+use isatty::stdout_isatty;
 use log::LevelFilter;
+use pretty_env_logger::env_logger::WriteStyle;
 use std::str::FromStr;
 
-pub fn setup_logger(log_config: &[LogValue]) {
+pub fn write_style_from_str(color: &str) -> WriteStyle {
+    match color {
+        "always" => WriteStyle::Always,
+        // resolve auto if stdout is available
+        "auto" => {
+            if stdout_isatty() {
+                WriteStyle::Always
+            } else {
+                WriteStyle::Never
+            }
+        }
+        "never" => WriteStyle::Never,
+        // this should never happen
+        // structopt would error before this catchall happens
```

Comment:
If this should never happen then perhaps, we should return an `Option` or a `Result` and handle the error and not silently ignore it? Or even unwrap? Or, use an enum, then no catchall would be needed.

Reply:
Going with the enum plan